constraints/src/main/java/gov/nasa/jpl/aerie/constraints/json/ConstraintParsers.java

```diff
@@ -152,6 +154,22 @@ static JsonParser<Rate> rateF(final JsonParser<Expression<LinearProfile>> linear
               untuple((kind, alias) -> new StartOf(alias)),
               $ -> tuple(Unit.UNIT, $.activityAlias)));
 
+  private static final JsonParser<Duration> durationP =
+      longP
+          . map(Iso.of(
+              microseconds -> Duration.of(microseconds, Duration.MICROSECONDS),
+              duration -> duration.in(Duration.MICROSECONDS)));
+
+  static JsonParser<ShiftBy> shiftByP(JsonParser<Expression<Windows>> windowsExpressionP) {
```

Comment:
:large_orange_diamond: I think this method should be named `shiftByF` in keeping with the naming convention.